pkg/client/client.go

```diff
@@ -1423,6 +1423,12 @@ func (c *Client) CreateIfNotExistSecret(ctx context.Context, s *v1.Secret) error
 	return errors.Wrap(err, "retrieving Secret object failed")
 }
 
+func (c *Client) CheckSecretExists(ctx context.Context, namespace, secretName string) error {
+	sClient := c.kclient.CoreV1().Secrets(namespace)
+	_, err := sClient.Get(ctx, secretName, metav1.GetOptions{})
+	return err
+}
+
```

Comment:
I think we should replace the call to `CheckSecretExists` with just the content of the function below since IMHO, it don't simplify the code. 
It may even be better to call the function `GetSecret(..) (*corev1.Secret, error)` and the caller can ignore the the secret returned.

Reply:
Thanks for that. I saw wee already have GetSecret() method so I have re-used that :)